### PR TITLE
Collect Errors

### PR DESCRIPTION
**Pull request by KyleKing**
Hi! I am implementing implements in a personal project and found that it was easier to collect all the errors at once. Not sure if you would be interested in adding this feature, but figured I would submit a PR

If interested, I would be happy to make any changes or code cleanup

### CODE REVIEW

**Comment by ksindi** [comment]
Interesting idea. Thanks @KyleKing!

File: implements.py [hunk 1]
```diff
@@ -44,9 +44,13 @@ def implements(interface_cls):
...
     def _decorator(cls):
-        verify_methods(interface_cls, cls)
-        verify_properties(interface_cls, cls)
-        verify_attributes(interface_cls, cls)
+        errors = []
+        errors.extend(verify_methods(interface_cls, cls))
+        errors.extend(verify_properties(interface_cls, cls))
+        errors.extend(verify_attributes(interface_cls, cls))
+        if len(errors):
+            raise NotImplementedError('Found errors in interfaces:\n'
```
Comment (pshirali) [inline implements.py]: Recommendation:
Change 'Found errors in interfaces:' to 'Found {} errors in implementation:', with '{}' formatted to print len(errors)

**Comment by pshirali** [comment]
@KyleKing this is a nice enhancement. Thank you for the contribution!

May I request a couple of things?
1. I have recommended a small [change to the error message](https://github.com/ksindi/implements/pull/21/commits/828f5ba0aae5ba9e0bc9e516b229b7846b0ac0d9#r510864385). I'll leave it to your discretion to implement it.
1. Could you please add one unittest which verifies that multiple errors are clubbed into a single error-msg. This may involve string matching, so you may choose to match just the minimum amount of string content; enough to verify that two or more errors are captured by the error msg. In case you agree to include my change, you could also verify against the count --- `{x}` in "Found {x} ..."

Thanks ! 👍

File: implements.py (hunk 1 above)
Comment (KyleKing) [reply implements.py]: I think the change is a good idea. Does this look good?

```py
raise NotImplementedError(
    'Found {} errors in implementation:\n- {}\nwith {}'.format(
        len(errors), '\n- '.join(errors), cls))
```

Outputs:

```
NotImplementedError: Found 2 errors in implementation:
- 'MallardDuck' must implement method 'fly(self) -> bool' defined in interface 'Quackable'
- 'MallardDuck' must implement method 'quack(self)' defined in interface 'Quackable'
with <class '__main__.MallardDuck'>
```

**Comment by KyleKing** [comment]
Thanks for the positive feedback! I think I implemented both changes you requested

**Comment by pshirali** [comment]
@KyleKing thank you so much! .. Approved 👍  .. Sorry for the delay in reviewing your changes.
@ksindi please feel free to merge.

**Pull request merged**